ipld/read.go

```diff
@@ -125,3 +148,10 @@ func GetLeavesByNamespace(
 	}
 	return out, err
 }
+
+// leafToShare convert an NMT leaf represented in IPLD into a Share
+func leafToShare(nd ipld.Node) Share {
+	// * First byte represents the type fo the node, and it's unrelated to the actual share data
+	// * Additional namespace is prepended so tha parity data can be identified with a parity namespace which we cut off
+	return nd.RawData()[1+NamespaceSize:] // TODO(@Wondertan): Rework NMT/IPLD plugin to avoid the type byte
```

Comment:
```suggestion
	return nd.RawData()[1+consts.NamespaceSize:] // TODO(@Wondertan): Rework NMT/IPLD plugin to avoid the type byte
```

Reply:
Shouldn't the same constant be used in all places?

Reply:
It is the same, because ‘NamespaceSize = consts.NamespaceSize’. I just redefined some of ipld pkg specific consts.

Reply:
Is there a reason to redefine the constant? Why not just use `consts.NamespaceSize` directly?

Reply:
@adlerjohn, @renaynay. The reason being is to make the `ipld` pkg self-sufficient, meaning that the user(we mostly) don't have to know about yet another dependency(app) to work with the API of this pkg/lib. This is just a general rule of thumb when you design go pkgs.

Reply:
Oh I see, that makes sense. Though I am concerned that if the value of the two constants drifts, then it's not immediately obvious. Would it be better to have some shared constants package?

Reply:
@adlerjohn,

The aliases are defined like this:
```
const (
	MaxSquareSize = consts.MaxSquareSize
	NamespaceSize = consts.NamespaceSize
	ShareSize = consts.ShareSize
)
```
and not like this:
```
const (
	MaxSquareSize = 128
	NamespaceSize = 8
	ShareSize = 8
)
```

This means that we still rely on the consts pkg by proxying values, and if something is drifting, we don't need to change things twice.